--- /usr/portage/games-emulation/dolphin/dolphin-9999.ebuild	2014-12-28 04:01:09.000000000 +0300
+++ dolphin-9999.ebuild	2015-01-08 02:30:11.410352771 +0300
@@ -26,7 +26,8 @@
 IUSE="alsa ao bluetooth doc ffmpeg +lzo openal opengl openmp portaudio pulseaudio"
 
 RDEPEND=">=media-libs/glew-1.10
-	>=media-libs/libsdl-1.2[joystick]
+	media-libs/libsdl[joystick]
+	media-libs/libsdl2[joystick,haptic]
 	>=media-libs/libsfml-2.1
 	>=net-libs/miniupnpc-1.8
 	sys-libs/readline
